runtime/runtime/src/congestion_control.rs

```diff
@@ -656,29 +676,61 @@ impl DelayedReceiptQueueWrapper {
         Ok(())
     }
 
+    // With ReshardingV3, it's possible for a chunk to have delayed receipts that technically belong
+    // the sibling shard before a resharding event.
+    // Here, we filter all the receipts that don't belong to the current shard_id.
+    //
+    // The function follows the guidelines of standard iterator filter function
+    // We return true if we should retain the receipt and false if we should filter it.
+    fn receipt_filter_fn(&self, receipt: &ReceiptOrStateStoredReceipt) -> bool {
+        if !ProtocolFeature::SimpleNightshadeV4.enabled(self.protocol_version) {
+            return true;
+        }
```

Comment:
Strictly speaking not required - I would remove it if it allows us to get rid of the protocol version from self. 

It may be interesting to assert that shard_id == self.shard_id before resharding V3.

Reply:
Yeah, we can get rid of the protocol_version check. I just thought it made sense to keep as much of the original behavior as possible, but will remove!